ionic/install_ubuntu_src.md

```diff
@@ -112,6 +112,19 @@ sudo apt -y install \
   $(sort -u $(find . -iname 'packages-'`lsb_release -cs`'.apt' -o -iname 'packages.apt' | grep -v '/\.git/') | sed '/gz\|sdf/d' | tr '\n' ' ')
 ```
 
+Alternatively, if you want to install dependencies using
+[rosdep](https://docs.ros.org/en/humble/Tutorials/Intermediate/Rosdep.html),
+ensure that [rosdep is installed](https://docs.ros.org/en/humble/Tutorials/Intermediate/Rosdep.html#how-do-i-use-the-rosdep-tool)
+and use the following command:
+
+```bash
+cd ~/workspace/src
+# gz-tools2 can build from source against both gz-cmake3 and gz-cmake4
+# this workspace has only gz-cmake4
+# the gz-tools2 package.xml only lists gz-cmake3, so use --skip-keys gz-cmake3
+rosdep install -i --from-path . --skip-keys gz-cmake3
```

Comment:
Have you checked if this works for the whole workspace? I don't think there are rosdep keys for `ogre-next` and `DART`.

Reply:
I'll test it now...

Reply:
it currently fails to resolve `DART` from the [gz-physics package.xml](https://github.com/gazebosim/gz-physics/blob/gz-physics8_8.0.0/package.xml#L16) and `libogre-next-2.3-dev` from the [gz-rendering9 package.xml](https://github.com/gazebosim/gz-rendering/blob/gz-rendering9_9.0.0/package.xml#L23)

I suppose we could skip those for now as well. I'll add comments and open issues in gz-physics and gz-rendering about how to name these dependencies in a way that both `colcon` and `rosdep` are happy with

Reply:
I believe we already have a solution, but it requires more steps. https://github.com/osrf/osrf-rosdep

Reply:
I just updated with the extra `--skip-keys` arguments and some explanation

> I believe we already have a solution, but it requires more steps. https://github.com/osrf/osrf-rosdep

I'm testing on `gLinux` with the ROS repos but not packages.osrfoundation.org. In this specific case, `osrf-rosdep` would help with resolving [DART to libdart-dev](https://github.com/osrf/osrf-rosdep/blob/master/gz/gazebo.yaml#L3-L4), so I can update the instructions to include it, but it wouldn't help with `gz-cmake3` or `libogre*` in this case